File: backend/src/main/java/ch/oliumbi/gyroscope/endpoints/incident/requests/IncidentRequest.java
package ch.oliumbi.gyroscope.endpoints.incident.requests;

import ch.oliumbi.gyroscope.core.incident.enums.IncidentSeverity;
import ch.oliumbi.gyroscope.core.incident.enums.IncidentStatus;
import ch.oliumbi.gyroscope.core.incident.enums.IncidentType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class IncidentRequest {
    private UUID assignee;
    private String title;
    private String system;
    private LocalDateTime time;
    private IncidentStatus status;
    private IncidentSeverity severity;
    private IncidentType type;
    private List<IncidentCheckRequest> checks;
}

File: frontend/src/requests/incident.request.ts
import {IncidentStatus} from "../enums/incident-status.enum.ts";
import {IncidentSeverity} from "../enums/incident-severity.enum.ts";
import {IncidentType} from "../enums/incident-type.enum.ts";
import {IncidentCheckRequest} from "./incident-check.request.ts";

export interface IncidentRequest {

    assignee: string,
    title: string,
    system: string,
    time: string,
    status: IncidentStatus,
    severity: IncidentSeverity,
    type: IncidentType,
    checks: IncidentCheckRequest[]
}
